### PR TITLE
LPD-34441 - Move filters form from modal to page

### PR DESCRIPTION
**Pull request by juanjofgliferay**
## Story / Task
[LPD-19693](https://liferay.atlassian.net/browse/LPD-19693) / [LPD-34441](https://liferay.atlassian.net/browse/LPD-34441)

## Goal
Main goal of the task is to move the components used to create and edit DSM Filters to a page instead of using a modal. 

### CX Filters
CX_filters

### Date Range Filters
DateRange_filters

### Selection Filters (API Headless)
selection_api_headless_filter

### Selection Filters (Picklist)
selection_picklist_filter

### CODE REVIEW

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx [hunk 1]
```diff
@@ -133,20 +140,24 @@ function AddFDSFilterModalContent({
 
 		onSave({...responseJSON, displayType, filterType});
 
-		closeModal();
+		dismiss();
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:143]: Do we need `dismiss` now? Can the setting of active mode now we just a part of `onSave`?

File: modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dateRangeFilters.spec.ts [hunk 2]
```diff
@@ -47,14 +49,23 @@ test.describe('Date range filter creation, edition and cancel', () => {
 	});
 
 	test.afterEach(async ({dataSetManagerApiHelpers}) => {
-		await dataSetManagerApiHelpers.deleteDataSet({erc: dataSetERC});
+
+		// await dataSetManagerApiHelpers.deleteDataSet({erc: dataSetERC});
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/dateRangeFilters.spec.ts:53]: :broom:

**Review by markocikos — COMMENTED** [review]
@juanjofgliferay few questions and comments inline

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/css/Filters.scss [hunk 3]
```diff
@@ -1,11 +1,14 @@
 @import 'cadmin-variables';
 
-html#{$cadmin-selector} .cadmin .dropdown-menu.fds-field-name-dropdown-menu {
+html#{$cadmin-selector} .cadmin .dropdown-menu.fds-field-name-dropdown-menu,
+html#{$cadmin-selector} .dropdown-menu.fds-field-name-dropdown-menu,
+html#{$cadmin-selector} .dropdown-menu.fds-entries-dropdown-menu {
```
[the remark this answers is not in the document]
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/css/Filters.scss:5]: I've found that the `.cadmin` class name appears when we open a modal. Basically, it creates a `<div class="cadmin"></div>` and injects the content of the Modal and all absolute positioned dropdowns related (this is the case of Data Sets page).

image

However, for the Filters page, moving from Modal to page, only creates a `<div>` without the `.cadmin` to hold the dropdowns.

image

Anyway, this is a leftover.

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx (hunk 1 above)
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:143]: Do you mean calling `setActiveMode(FILTER_MODE.LIST)` here instead?

Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:143]: Isn't `onSave` more natural, as `activeMode` is state in parent scope?

Both `dismiss` and `onSave` are callbacks on `FilterFormComponent`, executed here one after another. Is there a reason not to merge them?

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/css/Filters.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/css/Filters.scss:5]: Oh right, dropdowns do have multiple contexts.

Then, just a small simplification, we can leverage SASS with something like 
```sass
html#{$cadmin-selector} {
    .cadmin .dropdown-menu.fds-field-name-dropdown-menu,
    .dropdown-menu.fds-field-name-dropdown-menu,
    .dropdown-menu.fds-entries-dropdown-menu {
     }
}
```

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Comment by juanjofgliferay** [comment]
ci:test:sf

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx [hunk 4]
```diff
@@ -408,53 +402,139 @@ function Filters({
...
-				title={Liferay.Language.get('filters')}
-			/>
+			<ClayBreadcrumb className="my-2" items={getBreadcrumbItems()} />
+
+			{activeMode === FILTER_MODE.CREATION && (
+				<ClayLayout.Sheet>
+					{activeFilterType && (
+						<FilterFormComponent
+							dataSet={dataSet}
+							dismiss={() => setActiveMode(FILTER_MODE.LIST)}
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:455]: I still see a lot of `dismiss`es in this PR, 22 search hits. Can we get rid of all of them?

**Review by markocikos — COMMENTED** [review]
@juanjofgliferay just one question

File: modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx (hunk 4 above)
Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:455]: Well, they should handle the cancel button.

Comment (juanjofgliferay) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:455]: After removing this `dismiss` call 
image

maybe it makes more sense to rename the `dismiss` to `onCancel`

Comment (markocikos) [reply modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/data_set/filters/Filters.tsx:455]: `onCancel` sounds good, if the event is in child component and we cannot set `activeMode` inline.

**Comment by juanjofgliferay** [comment]
ci:test:sf

**Review by markocikos — APPROVED** [review]
@juanjofgliferay LGTM! :sparkles:

**Comment by juanjofgliferay** [comment]
ci:forward

**Comment by juanjofgliferay** [comment]
ci:test:relevant

**Comment by juanjofgliferay** [comment]
ci:forward

**Pull request closed without merging**